tests/core/kernel-and-base-single-reboot/task.yaml

```diff
@@ -95,14 +95,8 @@ execute: |
         change_id="$(cat refresh-change-id)"
         # XXX: is this sufficiently robust?
         snap watch "$change_id" || true
-        snap changes | MATCH "$change_id\s+(Done|Error)"
-        # we expect re-refresh to fail since the tests uses a fake store
-        snap change "$change_id" > tasks.done
-        MATCH '^Error .* Handling re-refresh' < tasks.done
-        # no other errors
-        grep -v 'Handling re-refresh' < tasks.done | NOMATCH '^Error'
-        # nothing was undone
-        grep -v 'Handling re-refresh' < tasks.done | NOMATCH '^Undone'
+        # we expect the change to complete
+        snap changes | MATCH "$change_id\s+Done"
```

Comment:
This was actually interesting, because it confused me to, I found out that because of the serializing of essential-snaps now, check-rerefresh actually ends up running last, and because check-rerefresh detects that all snaps have been refreshed as a part of the change, it actually ends up just returning and never contacting the fake-store.

Reply:
ah, check-rerefresh is not meant to be connected to other tasks because it needs work even if some task fail, which is not possible to achieve with actual liknks. From your description it sounds like now it's connected and we don't want that, this probably relates a bit to using WaitAll I suspect

Reply:
Finally figured this one out, "check-rerefresh" did not contact the store due to `refreshedSnaps` returning an empty list of refreshed snaps. It turns out that `refreshedSnaps` did not expect snaps to be in more than one lane (and thus just ignoring that snap), fixing this made the spread test behave like before.